### PR TITLE
Add create_mission_definition and start_mission_exicution

### PR DESCRIPTION
**Pull request by monkeyysups**

### CODE REVIEW

File: src/isar_exr/api/energy_robotics_api.py [hunk 1]
```diff
@@ -66,3 +66,46 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+            }}
+        """
+        params: dict = {
+            "site_id": site_id,
+            "mission_name": mission_name,
+            "exr_robot_id": exr_robot_id,
+        }
+
+        try:
+            response_dict: dict[str, Any] = self.client.query(mutation_string, params)
```
Comment (eivindsjovold) [inline src/isar_exr/api/energy_robotics_api.py:88]: is it sufficient to say dict?

File: src/isar_exr/api/energy_robotics_api.py [hunk 2]
```diff
@@ -66,3 +66,46 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+                }
+            }
+        """
+        params: dict = {"robot_id": robot_id, "mission_id": mission_id}
+
+        try:
+            response_dict: dict[str, Any] = self.client.query(mutation_string, params)
+        except Exception as e:
+            raise RobotException(e)
+        mission_id = response_dict["startMissionExecution"]["id"]
```
Comment (eivindsjovold) [inline src/isar_exr/api/energy_robotics_api.py:111]: should this be stored somewhere or returned?

File: src/isar_exr/api/energy_robotics_api.py [hunk 3]
```diff
@@ -66,3 +66,46 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+
+        try:
+            response_dict: dict[str, Any] = self.client.query(mutation_string, params)
+        except Exception as e:
+            raise RobotException(e)
+
+        mission_id = response_dict["createMissionDefinition"]["id"]
+        return mission_id
+
+    def star_mission_exicution(self, mission_id: str, robot_id: str) -> None:
```
Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:95]: Suggest renaming `mission_id` to `mission_definition_id` to be explicit

Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:95]:
```suggestion
    def start_mission_execution(self, mission_definition_id: str, robot_id: str) -> str:
```

File: src/isar_exr/api/energy_robotics_api.py [hunk 4]
```diff
@@ -66,3 +66,46 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+            "exr_robot_id": exr_robot_id,
+        }
+
+        try:
+            response_dict: dict[str, Any] = self.client.query(mutation_string, params)
+        except Exception as e:
+            raise RobotException(e)
+
+        mission_id = response_dict["createMissionDefinition"]["id"]
+        return mission_id
```
Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:93]: Suggest renaming `mission_id` to `mission_definition_id`

File: src/isar_exr/api/energy_robotics_api.py [hunk 5]
```diff
@@ -66,3 +66,46 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+                    id
+                }
+            }
+        """
+        params: dict = {"robot_id": robot_id, "mission_id": mission_id}
+
+        try:
+            response_dict: dict[str, Any] = self.client.query(mutation_string, params)
+        except Exception as e:
+            raise RobotException(e)
```
Comment (eivindsjovold) [inline src/isar_exr/api/energy_robotics_api.py:110]:
```suggestion
            raise RobotException from e
```

File: src/isar_exr/api/energy_robotics_api.py [hunk 6]
```diff
@@ -66,3 +66,46 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+        params: dict = {
+            "site_id": site_id,
+            "mission_name": mission_name,
+            "exr_robot_id": exr_robot_id,
+        }
+
+        try:
+            response_dict: dict[str, Any] = self.client.query(mutation_string, params)
+        except Exception as e:
+            raise RobotException(e)
```
Comment (eivindsjovold) [inline src/isar_exr/api/energy_robotics_api.py:90]:
```suggestion
            raise RobotException from e
```

File: src/isar_exr/api/energy_robotics_api.py (hunk 2 above)
Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:111]:
```suggestion
        mission_execution_id = response_dict["startMissionExecution"]["id"]
        return mission_execution_id 
```

**Review by GodVenn — CHANGES_REQUESTED** [review]
Some suggestions for naming and I think we will need to return the mission execution id

**Comment by GodVenn** [comment]
Can you add tests for these functions?
See the test for pause_current_mission for inspiration

File: src/isar_exr/api/energy_robotics_api.py (hunk 5 above)
Comment (monkeyysups) [reply src/isar_exr/api/energy_robotics_api.py:110]: done

File: src/isar_exr/api/energy_robotics_api.py (hunk 6 above)
Comment (monkeyysups) [reply src/isar_exr/api/energy_robotics_api.py:90]: done

File: src/isar_exr/api/energy_robotics_api.py (hunk 3 above)
Comment (monkeyysups) [reply src/isar_exr/api/energy_robotics_api.py:95]: done

File: src/isar_exr/api/energy_robotics_api.py (hunk 4 above)
Comment (monkeyysups) [reply src/isar_exr/api/energy_robotics_api.py:93]: done

File: src/isar_exr/api/energy_robotics_api.py (hunk 2 above)
Comment (monkeyysups) [reply src/isar_exr/api/energy_robotics_api.py:111]: done

Comment (monkeyysups) [reply src/isar_exr/api/energy_robotics_api.py:111]: sure, done

File: src/isar_exr/api/energy_robotics_api.py (hunk 1 above)
Comment (monkeyysups) [reply src/isar_exr/api/energy_robotics_api.py:88]: dont know what you mean. It's the same as the other functions we have made. Il just let it stand and we can update it later maby

File: tests/api/test_energy_robotics_api.py [hunk 7]
```diff
@@ -33,3 +33,63 @@ def test_fails_if_status_is_anything_else(self):
...
         with pytest.raises(expected_exception=Exception):
             api.pause_current_mission("test_exr_robot_id")
+
+
+@mock.patch(
+    "isar_exr.api.graphql_client.get_access_token",
+    mock.Mock(return_value="test_token"),
+)
+class Test_create_mission_definition:
+    create_mission_definition_response = "mission_definition"
```
Comment (GodVenn) [inline tests/api/test_energy_robotics_api.py:43]:
```suggestion
    expected_return_id = "mission_definition"
```

File: tests/api/test_energy_robotics_api.py [hunk 8]
```diff
@@ -33,3 +33,63 @@ def test_fails_if_status_is_anything_else(self):
...
+                exr_robot_id="mock_robot_id",
+            )
+
+
+@mock.patch(
+    "isar_exr.api.graphql_client.get_access_token",
+    mock.Mock(return_value="test_token"),
+)
+class Test_start_mission_execution:
+    create_mission_execution_response = "mission_execution_id"
```
Comment (GodVenn) [inline tests/api/test_energy_robotics_api.py:74]:
```suggestion
    expected_return_id = "mission_execution_id"
```

**Review by GodVenn — COMMENTED** [review]
Suggest renaming

File: tests/api/test_energy_robotics_api.py (hunk 7 above)
Comment (monkeyysups) [reply tests/api/test_energy_robotics_api.py]: done

File: tests/api/test_energy_robotics_api.py (hunk 8 above)
Comment (monkeyysups) [reply tests/api/test_energy_robotics_api.py]: done

**Review by GodVenn — APPROVED** [review]
LGTM

**Pull request merged**